app/controllers/documents/document_upload_question_controller.rb

```diff
@@ -9,30 +9,38 @@ def edit
       return if self.class.document_type.nil?
 
       @documents = current_intake.documents.of_type(document_type)
-      @form = form_class.new(document_type, current_intake, form_params)
+      @form ||= form_class.new(document_type, current_intake, form_params)
     end
 
     def update
       return if self.class.document_type.nil?
 
       @form = form_class.new(document_type, current_intake, form_params)
       if @form.valid?
-        form_saved = @form.save
+        @form.save
         after_update_success
         track_document_upload
+        if @form.try(:next_step)
+          redirect_to next_path and return
+        end
       end
 
-      redirect_to action: :edit
+      edit
```

Comment:
This is clearly not ideal and suggests that some larger refactoring is required such that instance variables required for rendering the edit template will be instantiated from all inherited controllers.